docs/proposals/20-ha-control-planes.md

```diff
@@ -0,0 +1,1259 @@
+* Across all `MachineDeployment` in the affected zone, one machine will be in `Terminating` state and other existing machines will be in `Unknown` state. There might one additional machine in `CrashLoopBackOff`.
+
+In case the `Machine-Controller-Manager` is able to delete the `Failed` machines then following will be observed:
+* For every `MachineDeployment` in the affected zone, there will be one machine stuck in `Pending` state and all other machines will be in `Unknown` state.
+* For the machine in `Pending` state there will not be any corresponding node object.
+* For all the machines in `Unknown` state, corresponding node resource will be in `NotReady/Unknown` state and there will be an additional taint `key: node.kubernetes.io/unreachable, effect: NoSchedule` on each of the node.
+
+If the above state is observed for an extended period of time (beyond a threshold that could be defined), then it can be deduced that there is a zone outage.
+
+### Identify zone recovery
+
+> NOTE: This section should be read in context of the current limited definition of a zone outage as described above.
+
+The machines which were previously stuck in either `Pending` or `CrashLoopBackOff` state are now in `Running` state and if there are corresponding `Node` resources created for machines, then the zonal recovery has started.
+
+> NOTE: In the future, when `Zonal Health` is introduced as a resource, then one could also inspect the status of this resource to identify zone recovery.
+
+### Future scope: Tracking Zone Health
+
+A zone is shared across one or more seed clusters. If an availability zone is no longer reachable or available then it can potentially impact more than one seed cluster. A request to schedule a new shoot control plane should avoid a non-available zone and therefore seed(s) which have their worker pools in unhealthy zones. This is especially valid for HA shoot control planes where control plane components like `etcd` (assuming a 3 member cluster) will then be provisioned with only 2 members, thereby running with no further failure tolerance.
+
+It is therefore proposed that a new custom resource `Zone` be introduced which will be initially used to track the health of a zone as part of its `status`. A `ZoneController` will periodically inspect the health of a zone by inspecting network connectivity, health of provisioned nodes, health of critical services hosted, etc. in the zone and report the state as part of its `status`. The consumer can leverage this information to further decide how to react.
+
+In general, tracking zonal health should be undertaken especially when the move is towards multi-zonal clusters. The above is just one proposal, but there can be other even better ways to achieve this. This is outside the purview of this GEP and should be taken up separately.
```

Comment:
I think we just wanted to sketch how we can further improve Gardener's HA capabilities and to make the reader aware of the gaps and open points we still have after implementing the proposal. Do you think it distracts too much?

Reply:
Ok, if you find it useful, then let's keep it! 👍

Reply:
You are right that it raises a lot of questions and answers little, but the original intention is to propose future work and provide a nudge in that direction so that we plan for next increments accordingly. It surely requires a separate GEP and in itself is a large topic if and when it is taken up.
@timuthy may be we can change the wording of this section. Right not it is somewhere in between.

Reply:
we have removed this section as this is not relevant in this version of the GEP. Future enhancement proposals will be separately raised where we will detail out the proposal properly.